libraries/ms-common/src/x86/gpio.c

```diff
@@ -12,7 +101,50 @@ static uint32_t prv_get_index(const GpioAddress *address) {
   return address->port * (uint32_t)NUM_GPIO_PORTS + address->pin;
 }
 
+static void prv_export() {
+  for (uint16_t i = 0; i < GPIO_TOTAL_PINS; i++) {
+    s_store.state[i] = s_pin_settings[i].state;
+  }
+  store_export(MX_STORE_TYPE__GPIO, &s_store, NULL);
+}
+
+static void update_store(ProtobufCBinaryData msg_buf, ProtobufCBinaryData mask_buf) {
+  MxGpioStore *msg = mx_gpio_store__unpack(NULL, msg_buf.len, msg_buf.data);
+  MxGpioStore *mask = mx_gpio_store__unpack(NULL, mask_buf.len, mask_buf.data);
+
+  for (uint16_t i = 0; i < mask->n_state; i++) {
+    // only update state if mask is set
+    if (mask->state[i] != 0) {
+      s_store.state[i] = msg->state[i];
+      if (s_pin_settings[i].state != (uint8_t)msg->state[i]) {
+        s_pin_settings[i].state = msg->state[i];
+        GpioAddress address = { .port = 0, .pin = i % 16 };
+        gpio_it_trigger_interrupt(&address);
+      }
+    }
+  }
+
+  mx_gpio_store__free_unpacked(msg, NULL);
+  mx_gpio_store__free_unpacked(mask, NULL);
+  prv_export();
+}
+
+static void prv_init_store(void) {
+  store_config();
+  StoreFuncs funcs = {
+    (GetPackedSizeFunc)mx_gpio_store__get_packed_size,
+    (PackFunc)mx_gpio_store__pack,
+    (UnpackFunc)mx_gpio_store__unpack,
+    (FreeUnpackedFunc)mx_gpio_store__free_unpacked,
+    (UpdateStoreFunc)update_store,
+  };
+  s_store.n_state = GPIO_TOTAL_PINS;
+  s_store.state = malloc(GPIO_TOTAL_PINS * sizeof(protobuf_c_boolean));
```

Comment:
I feel like we *technically* should have a free function in the `StoreFuncs` that frees this memory since this technically is a memory leak. Then again maybe not, and we should just document it real well that it's meant to be freed on process exit.

Reply:
Added comment in store.h `store_register()` mentioning they shouldn't be freed